src/index.js

```diff
@@ -43,6 +36,14 @@ class LambdaOffline {
           const invocationType = req.headers['x-amz-invocation-type'];
           const { functionName } = req.params;
 
+          const handlers = Object.keys(this.service.functions).reduce((acc, key) => {
+            const fun = this.service.getFunction(key);
+            const funOptions = functionHelper.getFunctionOptions(fun, key, servicePath, serviceRuntime);
```

Comment:
I think the function options should be only loaded when serverless offline runs the first time, and also we should create the handler when the lambda function needs to be invoked to hot reload it.

outside the route
```js
const funOptions = Object.keys(this.service.functions).reduce((acc, key) => {
      const fun = this.service.getFunction(key);
      acc[key] = functionHelper.getFunctionOptions(fun, key, servicePath, serviceRuntime);
      return acc;
    }, {});
```

into the route

```js
const options = funOptions[functionName];

if (!options) {
   return reply().code(404);
}

const handler = functionHelper.createHandler(options, {});
```

Reply:
Sounds good to me 👍 Thank you for looking through the PR!